packages/core/src/internal/temporal-util.ts

```diff
@@ -428,13 +428,20 @@ export function assertValidNanosecond (
   )
 }
 
-export function assertValidZoneId (fieldName: string, zoneId: string): void {
+const timeZoneValidityCache = new Map();
+
+export function assertValidZoneId(fieldName: string, zoneId: string): void {
+  let result = false;
+
   try {
-    Intl.DateTimeFormat(undefined, { timeZone: zoneId })
+    if (timeZoneValidityCache.get(zoneId)) return;
+    result = !!Intl.DateTimeFormat(undefined, { timeZone: zoneId });
```

Comment:
The linter is failing because of `packages/core/src/internal/temporal-util.ts:438:16: Unexpected object value in conditional. The condition is always true. (@typescript-eslint/strict-boolean-expressions)`

```suggestion
    result = Intl.DateTimeFormat(undefined, { timeZone: zoneId }) != null
```

Reply:
I quickly checked and `Intl.DateTimeFormat` always either returns a valid formatter or raises an error, so the case `formatter === null` cannot occur. Hence, we don't need to use the null check nor a double-negation cast.

Reply:
Fair point.